airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -22,69 +22,105 @@
+            client_id=config.lwa_app_id,
+            refresh_token=config.refresh_token,
+            host=self.endpoint.replace("https://", ""),
+        )
+        stream_kwargs = {
+            "url_base": self.endpoint,
+            "authenticator": auth,
+            "aws_signature": aws_signature,
+            "replication_start_date": config.replication_start_date,
+        }
+        return stream_kwargs
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
         try:
-            client.check_connection()
-            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
-        except Exception as e:
-            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
-
-    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
-        client = self._get_client(config)
-
-        return AirbyteCatalog(streams=client.get_streams())
-
-    def read(
-        self, logger: AirbyteLogger, config: Mapping[str, Any], catalog: ConfiguredAirbyteCatalog, state: MutableMapping[str, Any] = None
-    ) -> Generator[AirbyteMessage, None, None]:
-        client = self._get_client(config)
-
-        logger.info("Starting syncing Amazon Seller API")
-        for configured_stream in catalog.streams:
-            yield from self._read_record(logger=logger, client=client, configured_stream=configured_stream, state=state)
-
-        logger.info("Finished syncing Amazon Seller API")
-
-    @staticmethod
-    def _read_record(
-        logger: AirbyteLogger, client: BaseClient, configured_stream: ConfiguredAirbyteStream, state: MutableMapping[str, Any]
-    ) -> Generator[AirbyteMessage, None, None]:
-        stream_name = configured_stream.stream.name
-        is_report = client._amazon_client.is_report(stream_name)
-
-        if configured_stream.sync_mode == SyncMode.full_refresh:
-            state.pop(stream_name, None)
-
-        if is_report:
-            yield from client.read_reports(logger, stream_name, state)
-        else:
-            yield from client.read_stream(logger, stream_name, state)
+            config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
+            stream_kwargs = self._get_stream_kwargs(config)
+            merchant_listings_reports_gen = MerchantListingsReports(**stream_kwargs).read_records(sync_mode=SyncMode.full_refresh)
+            next(merchant_listings_reports_gen)
+            return True, None
+        except Exception as error:
+            return False, f"Unable to connect to Amazon Seller API with the provided credentials - {repr(error)}"
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
```

Comment:
Well, for now we get config as a common python dict, it would be really good if it was `ConnectorConfig` object.
I'm not sure if we have such issue now.